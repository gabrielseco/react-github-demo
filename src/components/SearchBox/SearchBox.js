// @flow
import React, { Component } from 'react';
import styles from './SearchBox.scss';

type Props = {
  classesInput: string,
  placeholder: string,
  onChange: () => void
};

class SearchBox extends Component<Props> {
  constructor(props: Props) {
    super(props);
    this.state = {};
    this.onChange = this.onChange.bind(this);
  }

  onChange(evt) {
    if (this.props.onChange) {
      this.props.onChange({
        value: evt.target.value
      });
    }
  }

  render() {
    const { classesInput, placeholder } = this.props;

    return (
      <div className={styles.container}>
        <input
          className={`${classesInput} search-box`}
          type="text"
          placeholder={placeholder}
          onChange={this.onChange}
        />
      </div>
    );
  }
}

SearchBox.defaultProps = {
  classesInput: '',
  placeholder: '',
  onHandleTyping: () => {}
};

export default SearchBox;
