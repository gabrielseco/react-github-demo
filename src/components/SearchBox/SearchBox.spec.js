import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SearchBox from './SearchBox';

describe('SearchBox suite', () => {
  it('renders SearchBox without any state injected', () => {
    const component = shallow(<SearchBox />);
    const tree = renderer.create(<SearchBox />).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
