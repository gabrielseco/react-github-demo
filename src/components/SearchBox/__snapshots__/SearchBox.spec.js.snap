// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox suite renders SearchBox without any state injected 1`] = `
<div
  className="container"
>
  <input
    className=" search-box"
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</div>
`;
